pkgs/development/libraries/mesa/default.nix

```diff
@@ -176,6 +181,8 @@ in stdenv.mkDerivation {
   # Needed to discover llvm-config for cross
   preConfigure = ''
     PATH=${llvmPackages.libllvm.dev}/bin:$PATH
+  '' + lib.optionalString (stdenv.targetPlatform.useLLVM or false) ''
+    export PKG_CONFIG_PATH_FOR_BUILD="${glslang}/lib/pkgconfig:$PKG_CONFIG_PATH_FOR_BUILD"
```

Comment:
Wait, doesn't this conflict with https://github.com/NixOS/nixpkgs/pull/330065?

Reply:
It would

Reply:
Now I'm thinking, would moving all libs out of the `$out` in glslang into `$dev` fix needing this silly override?

Reply:
Ok, splitting it out didn't fix it still but putting `spirv-tools` inside of `depsBuildBuild` & removing `glslang` from `buildInputs` seemed to have fixed it.

Reply:
What is spriv-tools used for? I don't imagine it'd generate any code for the buildPlatform?

Reply:
intel-clc

Reply:
For what I gathered, intel-clc is a OpenCL compiler. How is spirv-tools used "for intel-clc" here? At runtime?

Reply:
There's a set of libraries in spirv-tools that get linked to intel-clc.

Reply:
In that case the correct place to put it should be depsHostTarget aka. buildInputs.

Reply:
Why would it be trying to use `buildPackages.glslang`? `glslang` is in `buildInputs` and `strictDeps = true;`; it should be built from host to target which corresponds with the regular `pkgs`.

Reply:
Because pkg-config is preferring the `PKG_CONFIG_PATH_FOR_BUILD`

Reply:
It looks like separating `glslang` to having 3 outputs (`out`, `bin`, and `dev`) and explicitly specifying `glslang.bin` in `nativeBuildInputs` while having `spirv-tools` in `buildInputs` works.